spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultPollableMessageSource.java

```diff
@@ -204,6 +209,11 @@ public boolean poll(MessageHandler handler, ParameterizedTypeReference<?> type)
 
 		AcknowledgmentCallback ackCallback = StaticMessageHeaderAccessor
 				.getAcknowledgmentCallback(message);
+
+		if (ackCallback == null) {
+			ackCallback = status -> log.warn("No AcknowledgementCallback defined. Status: " + status.name() + " " + message);
+		}
```

Comment:
Wouldn't it be better to fix the `TestChannelBinder` as well?

Reply:
I'll take a look

Reply:
I'm not sure how to fix `TestChannelBinder` . It is the responsibility of the `MessageSource` to set the header. TCB doesn't enter into it AFAICT.  I could override `bindPollableConsumer` if you think it's necessary.

Reply:
?

```java
	private MessageSource<?> messageSourceDelegate = () -> new GenericMessage<>(
			"polled data",
			Collections.singletonMap(MessageHeaders.CONTENT_TYPE, "text/plain"));
```

Just add a no-op ack to the canned message that the test binder's message source emits.

Reply:
Ok. I wasn't thinking about that since my test overrides the message source.  The custom message source can be configured to cause an exception  - how I discovered this issue in the retry loop. I'm not sure that this one fails. I'll take another look.